### PR TITLE
Raise when downloaded package is not an agent package on fetch command

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR updates the agent fetching logic to raise proper exception when a downloaded package is not an agent package

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/fetch.py [hunk 1]
```diff
@@ -268,3 +277,7 @@ def fetch_mixed(
             f"Fetch from local registry failed (reason={str(e)}), trying remote registry..."
         )
         fetch_agent(ctx, public_id, alias=alias, target_dir=target_dir)
+
+
+class NotAnAgentPacakge(Exception):
```
Comment (dvilelaf) [inline aea/cli/fetch.py:282]: Typo: Package

File: aea/cli/fetch.py [hunk 2]
```diff
@@ -159,6 +162,12 @@ def fetch_agent_ipfs(
 
     ctx.clean_paths.append(target_dir)
     ctx.cwd = str(target_dir)
+
+    if not Path(target_dir, DEFAULT_AEA_CONFIG_FILE).exists():
+        raise NotAnAgentPacakge(
```
Comment (dvilelaf) [inline aea/cli/fetch.py:167]: Typo: Package

File: aea/cli/fetch.py [hunk 3]
```diff
@@ -76,7 +76,10 @@ def fetch(
     """Fetch an agent from the registry."""
     ctx = cast(Context, click_context.obj)
     ctx.registry_type = registry
-    do_fetch(ctx, public_id, alias)
+    try:
+        do_fetch(ctx, public_id, alias)
+    except NotAnAgentPacakge as e:
```
Comment (dvilelaf) [inline aea/cli/fetch.py:81]: Typo: Package

**Review by dvilelaf — COMMENTED** [review]
LGTM, waiting for CI

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**